src/main/java/net/minecraftforge/common/util/ITeleporter.java

```diff
@@ -19,49 +19,80 @@
-
+import net.minecraft.block.PortalInfo;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.player.PlayerEntity;
 import net.minecraft.entity.player.ServerPlayerEntity;
-import net.minecraft.world.Teleporter;
+import net.minecraft.util.TeleportationRepositioner;
+import net.minecraft.util.math.vector.Vector3d;
 import net.minecraft.world.server.ServerWorld;
 
+import java.util.Optional;
+import java.util.function.Function;
+
 /**
  * Interface for handling the placement of entities during dimension change.
- *
+ * <p>
  * An implementation of this interface can be used to place the entity
  * in a safe location, or generate a return portal, for instance.
- *
+ * <p>
  * See the {@link net.minecraft.world.Teleporter} class, which has
  * been patched to implement this interface, for a vanilla example.
  */
-public interface ITeleporter {
-
+public interface ITeleporter
+{
     /**
      * Called to handle placing the entity in the new world.
-     *
+     * <p>
      * The initial position of the entity will be its
      * position in the origin world, multiplied horizontally
      * by the computed cross-dimensional movement factor.
-     *
+     * <p>
      * Note that the supplied entity has not yet been spawned
      * in the destination world at the time.
      *
-     * @param entity           the entity to be placed
-     * @param currentWorld     the entity's origin
-     * @param destWorld        the entity's destination
-     * @param yaw              the suggested yaw value to apply
+     * @param entity the entity to be placed
+     * @param currentWorld the entity's origin
+     * @param destWorld the entity's destination
+     * @param yaw the suggested yaw value to apply
      * @param repositionEntity a function to reposition the entity, which returns the new entity in the new dimension. This is the vanilla implementation of the dimension travel logic. If the supplied boolean is true, it is attempted to spawn a new portal.
+     *
      * @return the entity in the new World. Vanilla creates for most {@link Entity}s a new instance and copy the data. But <b>you are not allowed</b> to create a new instance for {@link PlayerEntity}s! Move the player and update its state, see {@link ServerPlayerEntity#changeDimension(net.minecraft.world.server.ServerWorld, ITeleporter)}
      */
-    default Entity placeEntity(Entity entity, ServerWorld currentWorld, ServerWorld destWorld, float yaw, Function<Boolean, Entity> repositionEntity) {
-       return repositionEntity.apply(true);
+    default Entity placeEntity(Entity entity, ServerWorld currentWorld, ServerWorld destWorld, float yaw,
+            Function<Boolean, Entity> repositionEntity)
+    {
+        return repositionEntity.apply(true);
     }
-    
-    //used internally to handle vanilla hardcoding
+
+    /**
+     * Is this teleporter the vanilla instance.
+     */
     default boolean isVanilla()
     {
-        return getClass() == Teleporter.class;
```

Comment:
Why was this changed as it adds an extra line in the patch to Teleporter.java

Reply:
I figured returning true/false was more performant than checking for class equality.

Reply:
But I don't know java well enough to know if it makes a difference.

Reply:
Fixed